yt_dlp/downloader/common.py

```diff
@@ -335,6 +335,16 @@ def _report_progress_status(self, s, default_template):
             progress_template.get('download-title') or 'yt-dlp %(progress._default_template)s',
             progress_dict))
 
+        percent = s.get('_percent')
+        if s['status'] not in ('downloading', 'error', 'finished') or percent is None:
+            return
```

Comment:
Why this check? Where did you get these states? I don't remember there being an error state...

Reply:
https://github.com/yt-dlp/yt-dlp/blob/9f42e68a74f3f00b0253fe70763abd57cac4237b/README.md?plain=1#L2067-L2070
https://github.com/yt-dlp/yt-dlp/blob/9f42e68a74f3f00b0253fe70763abd57cac4237b/yt_dlp/YoutubeDL.py#L378-L382

I was following "Check this first and ignore unknown values". If it's inaccurate it should be adjusted first. `started`, as suggested below, is also none of the values advertised, which means it should be ignored.

Reply:
Seems I forgot to update docs when hooks were added to PP.

As for "ignore unknown values", it's for future-proofing and only applies to third party code

Reply:
I see. I dont know these very well, so thats why I followed the docs lmao

Reply:
So, I just had a quick look through code and only found `finished`, `downloading`, `started`. Maybe `error` was planned to be added but never finished? The docs are from ytdl.

We might as well keep your support for error state though.